impeller/scene/node.h

```diff
@@ -33,6 +34,11 @@ class Node final {
   Node(Node&& node);
   Node& operator=(Node&& node);
 
+  const std::string& GetName() const;
+  void SetName(const std::string& new_name);
+
+  std::shared_ptr<Node> FindNodeByName(const std::string& name) const;
```

Comment:
I didn't see the usage of FindNodeByName and can't tell if it matters that the lookup is linear with a string search. Its probably fine though.

Reply:
This is used for animation channel binding when a new `AnimationClip` is created in https://github.com/flutter/engine/pull/38595 (renamed to `FindChildByName`).

I haven't finished thinking these through, but a few options for improvement down the road:
* If we keep this name-only lookup, we could capture the common case by pre-generating binding state at deserialization time which will get used only if the user applies the animation to the deserialized node it came from (probably overwhelmingly common).
* Another option (which I like better) would be to form channel lookup paths at deserialization time and store them in the `Animation` instead of flat node names. This makes Animation retargeting less forgiving, but also makes name collisions less likely and prevents us from having to walk the whole scene for every channel. We could even add more advanced pathing features later like globbing or indexing.